.github/workflows/ci.yaml

```diff
@@ -1,73 +1,58 @@
-      - name: Test
+      - name: make-test
+        if: success()
         run: make test && make checkgenerate
-      - name: Lint
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - name: setup-go
+        if: success()
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.x
+      - name: make-lint
+        if: success()
         run: make lint
-      - name: GCP Auth
-        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
-        uses: google-github-actions/auth@v0
+  docker-build-push-connect-demo:
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs:
+      - test
+      - lint
+    steps:
+      - name: setup-docker-buildx
+        uses: docker/setup-buildx-action@v2
+      - name: login-gcr
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.PRIVATE_REGISTRY_HOST }}
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.PRIVATE_DOCKER_TOKEN }}
+      - name: build-push
+        uses: docker/build-push-action@v3
         with:
-          credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
-      - name: Set up GCP SDK
-        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
-        uses: google-github-actions/setup-gcloud@v0
-      - name: Authorize Docker push
-        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
-        run: gcloud auth configure-docker ${{ env.REGION }}-docker.pkg.dev
-      - name: Build and Push Container
-        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
-        run: |-
-          docker build -t ${{ env.REGION }}-docker.pkg.dev/${{ secrets.GCP_PROJECT }}/${{ env.REGISTRY }}/${{ env.SERVICE }}:${{ github.sha }} .
-          docker push ${{ env.REGION }}-docker.pkg.dev/${{ secrets.GCP_PROJECT }}/${{ env.REGISTRY }}/${{ env.SERVICE }}:${{ github.sha }}
-      - name: Deploy to Cloud Run
-        if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
-        run: |-
-          gcloud run deploy ${{ env.SERVICE }} \
-          --image=${{ env.REGION }}-docker.pkg.dev/${{ secrets.GCP_PROJECT }}/${{ env.REGISTRY }}/${{ env.SERVICE }}:${{ github.sha }} \
-          --port=8080 \
-          --concurrency=512 \
-          --cpu=2 \
-          --ingress=all \
-          --max-instances=5 \
-          --min-instances=default \
-          --memory=512Mi \
-          --platform=managed \
-          --timeout=5m \
-          --use-http2 \
-          --allow-unauthenticated \
-          --cpu-throttling \
-          --region=${{ env.REGION }} \
-          --quiet
+          file: Dockerfile
+          push: true
+          tags: |
+            ${{ env.PRIVATE_REGISTRY_URL }}/connect-demo:latest
+            ${{ env.PRIVATE_REGISTRY_URL }}/connect-demo:${{ github.sha }}
```

Comment:
This pushes the image; where and how does the new image get deployed, and how will we know if it fails?

Reply:
The deploy has to go through terraform. Unfortunately CloudRun needs the image SHA specified, and you need a new revision to change it. There's no good way around it. The alternatives would be substantially more overhead (eg maintaining another GKE cluster).

Reply:
That's pretty unpleasant. It's not a limitation present in the `gcloud` CLI or the Cloud Run UI, so I assume it comes from Terraform?

Reply:
Somewhat - that's _technically_ a mix of both. Both `gcloud` and the `Cloud Run UI` _won't_ automatically deploy when a new container image is available - you have to manually nudge them (eg running `gcloud` again).

We don't currently have a way of setting our Terraform to auto run on events.

I'll try to hook up something that will be automated on releases.

Reply:
They don't deploy when a new image is available, but they also don't need a SHA - they're happy to deploy `:latest`, which is easy to understand and doesn't require extra boilerplate PRs.

We're not deploying this all the time, so it's not worth sinking a ton of time into making this super-frictionless. I do appreciate all the work you're putting into migrating us to proper, company hosted infra! I'm just a little sad that, at least IMO, the overall developer experience gets worse when we're on TF.